### DISCUSSION TITLE
Is it possible to register a customization with a fixture by runtime types?

### DISCUSSION THREAD

**Discussion by kharabasz** [AutoFixture/AutoFixture#1447] [Q&A] [answered]
I know that it is possible to resolve a type from a fixture using a runtime type:
```
new SpecimenContext(fixture).Resolve(type);
```

Is it possible to register a customization with a fixture by runtime types?

```
fixture.Register(interfaceType, () => new SpecimenContext(fixture).Resolve(implementationType));
```

**Comment by Kralizek** [comment, marked as answer]
You can use a TypeRelay to make sure that all request of a given non-concrete type (interface or abstract class) is resolved with a specific concrete type.

For example:

```csharp
new TypeRelay(typeof(IList<>), typeof(List<>))
```

**Comment by behdad088** [comment]
Hi @kharabasz,
As @Kralizek  mentioned, you can use [TypeRelay](https://autofixture.io/api/AutoFixture.Kernel.TypeRelay.html)
